src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_cpu_blocking.cpp

```diff
@@ -56,14 +56,35 @@ LinearIR::constExprIt BrgemmCPUBlocking::get_loop_begin_pos(LinearIR& linear_ir,
     return loop_begin_it;
 }
 
-std::tuple<size_t, size_t, size_t> BrgemmCPUBlocking::get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr) {
-    auto blocking_params = ov::snippets::lowered::pass::BrgemmBlockingBase::get_blocking_params(brgemm_expr);
+std::tuple<size_t, size_t, size_t> BrgemmCPUBlocking::get_blocking_params(const ov::snippets::lowered::ExpressionPtr& brgemm_expr) const {
     const auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(brgemm_expr->get_node());
-    if (with_repacking(brgemm->get_type())) {
-        std::get<1>(blocking_params) = get_full_dim_value();
-        std::get<2>(blocking_params) = get_full_dim_value();
-    }
-    return blocking_params;
+    OPENVINO_ASSERT(brgemm, "BrgemmCPU is expected!");
+
+    size_t m, n, k;
+    std::tie(m, n, k) = get_brgemm_dimensions(brgemm_expr);
+
+    // Ticket: 113745
+    // TODO: extend block size selection heuristics
+    auto get_block_size_m = [](const size_t M) -> size_t {
+        const size_t default_m_blk = 32;
+        if (!snippets::utils::is_dynamic_value(M) && M <= default_m_blk)
+            return get_full_dim_value();
+        return default_m_blk;
+    };
+    auto get_block_size_n = [&](const size_t N) -> size_t {
+        const size_t default_n_blk = dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core) ? 64 : 24;
+        if (with_repacking(brgemm->get_type()) || (!snippets::utils::is_dynamic_value(N) && N <= default_n_blk))
+            return get_full_dim_value();
+        return default_n_blk;
+    };
+    auto get_block_size_k = [&](const size_t K) -> size_t {
+        const size_t high_k_blk = 1024;
+        const size_t low_k_blk = 512;
+        if (with_repacking(brgemm->get_type()) || (!ov::snippets::utils::is_dynamic_value(K) && K <= low_k_blk))
+            return get_full_dim_value();
+        return !ov::snippets::utils::is_dynamic_value(K) && K > high_k_blk ? high_k_blk : low_k_blk;
+    };
```

Comment:
Since we have base implementation as well, it would be much easier to understand these changes if they were incremental. Imagine we need to compare blocking params of BrgemmTPP vs BrgemmCPU: we would have to analyze the whole method (although the lambdas are very similar, but still not identical) instead of analyzing incremental changes only.
For example: `get_block_size_m` is idential to the base impl, but have an extra variable `default_m_blk`, `get_block_size_n` is also almost identical, but has hariable `default_n_blk` and covers the `with_repacking` case.
So my suggestion:
1) Move `default_m_blk`, `default_n_blk`, `default_k_blk` to the base class and, if necessary, change their value based on the isa in the constructor.
2) Rewrite the function in an incremental manner:
```
// get base values
std::tie(m_blk, n_blk, k_blk) = BrgemmBlockingBase::get_blocking_params(...);
// cover additional corner cases
if (with_repacking(brgemm->get_type()) {
    n_blk = get_full_dim_value();
    k_blk = get_full_dim_value();
}
```
Do you think we can do it to make the blocking logic more clear?
If not, then it probably doesn't make much sense to keep the base implementation.